src/traits.rs

```diff
@@ -283,6 +283,30 @@ where
                         }
                     }
                 }
+                Target::AllExcept(known) => {
```

Comment:
I wouldn't call the argument `known`: It's not necessarily all nodes known to the algorithm. Maybe `exclude`?

Reply:
Will change it to `exclude`.